<launch>
	<arg name="image" default="camera/rgb/image_color"/>
	<arg name="depth_image" default="camera/depth/image"/>
	<arg name="camera_topic" default="/camera/rgb/camera_info"/>


  <node name="cmvision" pkg="cmvision" type="cmvision" args="image:=$(arg image)" output="screen">
	  <!-- Location of the cmvision color file -->
	  <param name="color_file" type="string" 
	         value="$(find cmvision_3d)/colors/example.txt" />

	  <!-- Turn debug output on or off -->
	  <param name="debug_on" type="bool" value="false"/>

	  <!-- Turn color calibration on or off -->
	  <param name="color_cal_on" type="bool" value="false"/>

	  <!-- Enable Mean shift filtering -->
	  <param name="mean_shift_on" type="bool" value="true"/>

	  <!-- Spatial bandwidth: Bigger = smoother image -->
	  <param name="spatial_radius_pix" type="double" value="2.0"/>

	  <!-- Color bandwidth: Bigger = smoother image-->
	  <param name="color_radius_pix" type="double" value="40.0"/>
  </node>

  <node name="color_controller" pkg="cmvision_3d" type="color_controller.py" output="screen">
  		<param name="depth_image" value="$(arg depth_image)"/>
  		<param name="camera_topic" value="$(arg camera_topic)"/>
  		<param name="parent_frame" value="/map"/>
  </node>

  <!-- <node name="color_broadcaster" pkg="cmvision_3d" type="color_broadcaster.py" output="screen"/> -->

</launch>
